pkg/plugins/golang/v2/scaffolds/api.go

```diff
@@ -92,8 +79,15 @@ func (s *apiScaffolder) newUniverse() *model.Universe {
 }
 
 func (s *apiScaffolder) scaffold() error {
-	if s.doResource {
-		s.config.UpdateResources(s.resource.GVK())
+	// Check if we need to do the API and controller before updating with previously existing info
+	doAPI := s.resource.API != nil && s.resource.API.Version != ""
```

Comment:
Why we should scaffold the resource when the `s.resource.API.Version != ""` ?
IMO we need here respected the info / flag informed If the input is --resource/doApi as it was before.

Reply:
Also, see that we ought to not change the behaviour for v2. So why we are doing these changes here and increasing teh risk to introduce a bug ?